tty/usbacm/usbacm.c

```diff
@@ -58,32 +58,42 @@
 
 #define USBACM_BULK_SZ 2048
 
+/* clang-format off */
+#define TRACE(fmt, ...) if (0) printf("usbacm: " fmt "\n", ##__VA_ARGS__)
+
+enum { RxStopped = 0, RxRunning = 1, RxDisconnected = -1 };
+/* clang-format on */
 
 typedef struct _usbacm_dev {
 	struct _usbacm_dev *prev, *next;
-	int rxtid;
-	char path[32];
 	usb_devinfo_t instance;
+
+	/* USB HOST related */
 	int pipeCtrl;
 	int pipeIntIN;
 	int pipeBulkIN;
 	int pipeBulkOUT;
-	int id;
-	int fileId;
+	int urbs[USBACM_N_URBS];
+	int urbctrl;
+
+	char path[32];
+	unsigned int id; /* ACM id (/dev/usbacm[ID]) */
+	int fileId;      /* oid.id */
+
+	/* opened file state */
 	int flags;
 	pid_t clientpid;
-	volatile int rfcnt;
-	volatile int thrFinished;
-	unsigned port;
-	fifo_t *fifo;
+
+	volatile int rfcnt; /* protected by usbacm_common.lock */
+
+	/* blocking/nonblocking READ state - protected by rxLock */
 	char *rxptr;
-	int rxrunning;
 	size_t rxbuflen;
-	oid_t oid;
+	fifo_t *fifo;
 	handle_t rxLock;
 	handle_t rxCond;
-	int urbs[USBACM_N_URBS];
-	int urbctrl;
+	int rxState;
```

Comment:
As it's widely used without a lock
```suggestion
	volatile int rxState;
```

Reply:
changed always under lock, made it more explicit